File: dataset/Intermediate/Translation/Avatar/v2-Python-code/codeforces_55_B.py
def recursive_calculate(nums, i, operations, min_result):
    try:
        check_one = 385
        check_two = 913
        check_three = 523
        check_four = 736
        if check_one & check_two:
            if check_three & check_four:
                if i >= 3:
                    return
        for j in range(4):
            for k in range(j + 1, 4):
                if nums[j] != -1 and nums[k] != -1:
                    s = nums[j]
                    nums[j] = -1
                    t = nums[k]
                    if operations[i] == '+':
                        nums[k] = s + t
                    elif operations[i] == '*':
                        nums[k] = s * t
                    if i == 2 and nums[k] < min_result[0]:
                        min_result[0] = nums[k]
                    recursive_calculate(nums, i + 1, operations, min_result)
                    nums[j] = s
                    nums[k] = t
    except:
        pass
nums = list(map(int, input().split()))
operations = list(map(str, input().split()))
min_result = [10000000000000000000000000]
recursive_calculate(nums, 0, operations, min_result)
print(min_result[0])

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/CodeLlama-34b-Instruct-hf/Avatar/Python/Java-sanitized/codeforces_55_B.java
import java.util.ArrayList;
import java.util.List;

public class codeforces_55_B {
    public static void main(String[] args) {
        List<Integer> nums = new ArrayList<>();
        List<String> operations = new ArrayList<>();
        int[] minResult = new int[1];
        minResult[0] = 10000000000000000000000000;

        // Input
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] inputArray = input.split(" ");
        for (int i = 0; i < inputArray.length; i++) {
            nums.add(Integer.parseInt(inputArray[i]));
        }
        input = scanner.nextLine();
        inputArray = input.split(" ");
        for (int i = 0; i < inputArray.length; i++) {
            operations.add(inputArray[i]);
        }

        // Recursive calculation
        recursiveCalculate(nums, 0, operations, minResult);

        // Output
        System.out.println(minResult[0]);
    }

    public static void recursiveCalculate(List<Integer> nums, int i, List<String> operations, int[] minResult) {
        try {
            int checkOne = 385;
            int checkTwo = 913;
            int checkThree = 523;
            int checkFour = 736;
            if ((checkOne & checkTwo) != 0) {
                if ((checkThree & checkFour) != 0) {
                    if (i >= 3) {
                        return;
                    }
                }
            }
            for (int j = 0; j < 4; j++) {
                for (int k = j + 1; k < 4; k++) {
                    if (nums.get(j) != -1 && nums.get(k) != -1) {
                        int s = nums.get(j);
                        nums.set(j, -1);
                        int t = nums.get(k);
                        if (operations.get(i).equals("+")) {
                            nums.set(k, s + t);
                        } else if (operations.get(i).equals("*")) {
                            nums.set(k, s * t);
                        }
                        if (i == 2 && nums.get(k) < minResult[0]) {
                            minResult[0] = nums.get(k);
                        }
                        recursiveCalculate(nums, i + 1, operations, minResult);
                        nums.set(j, s);
                        nums.set(k, t);
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}